src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs

```diff
@@ -200,7 +201,12 @@ public void Dispose()
                     if (!_process.HasExited)
                     {
                         _process.Kill();
-                        _process.WaitForExit();
+                        _process.WaitForExit(processExitTimeoutInMilliseconds);
+
+                        if (!_process.HasExited)
+                        {
```

Comment:
do not throw.  `WaitForExit` returns a bool . Check the result. If `false`, log warning and call `process.Kill()`